python/tests/integration/arcticdb/test_arctic_additional.py

```diff
@@ -0,0 +1,421 @@
+    lib.write(symbol,df2, staged=True)
+    lib.write(symbol,df3, staged=True)
+    lib.finalize_staged_data(symbol=symbol,mode=StagedDataFinalizeMethod.APPEND)
+
+    result:pd.DataFrame = lib.read(symbol).data
+    dataframe_dump_to_log("RESULT DF:", result)
+
+    verify_dataframe_column(df=result, row_name="NUMBER0", max_type=last_type_a, array=arr_all_a)
+    verify_dataframe_column(df=result, row_name="NUMBER1", max_type=last_type_b, array=arr_all_b)
+    verify_dataframe_column(df=result, row_name="NUMBER2", max_type=last_type_c, array=arr_all_c)
+    verify_dataframe_column(df=result, row_name="NUMBER3", max_type=last_type_str, array=arr_all_str)
+    verify_dataframe_column(df=result, row_name="NUMBER5", max_type=last_type_d, array=arr_all_d)
+    verify_dataframe_column(df=result, row_name="NUMBER4", max_type=last_type_bool, array=arr_all_bool)
+
+    assert 3 == len(lib.list_versions(symbol=symbol))
+
+    lib.delete(symbol=symbol,versions=2)
+
+    assert 2 == len(lib.list_versions(symbol=symbol))
+
+    result:pd.DataFrame = lib.read(symbol).data
+    dataframe_dump_to_log("RESULT DF:", result)
+
+    #We want to validate that after delete last version the empty cell are 
+    #filled with 0 not nans
+    verify_dataframe_column(df=result, row_name="NUMBER1", max_type=np.int16, array=concat_all_arrays(arr_b1, arr_b2))
+
+    # As final step we once again repeat staging op with same data
+    # To arrive at final stage
+    lib.write(symbol,df2, staged=True)
+    lib.write(symbol,df3, staged=True)
+    lib.finalize_staged_data(symbol=symbol,mode=StagedDataFinalizeMethod.APPEND)
+
+    result:pd.DataFrame = lib.read(symbol).data
+    dataframe_dump_to_log("RESULT DF:", result)
+    assert 3 == len(lib.list_versions(symbol=symbol))
+
+    #Some final confirmations all is ok
+    verify_dataframe_column(df=result, row_name="NUMBER1", max_type=last_type_b, array=arr_all_b)
+    verify_dataframe_column(df=result, row_name="NUMBER2", max_type=last_type_c, array=arr_all_c)
+
+
+def test_finalize_staged_data_long_scenario(arctic_library_lmdb):
+    """
+        The purpose of of the test is to assure all staged segments along with their data
+        are correctly finalized and resulting
+    """
+
+    lib = arctic_library_lmdb
+    symbol = "symbol"
+
+    cachedDF = CachedDFGenerator(25000, [1])
+
+    total_number_rows: TimestampNumber = TimestampNumber(0, cachedDF.TIME_UNIT) # Synchronize index frequency
+
+    num_rows_initially = 99
+    print(f"Writing to symbol initially {num_rows_initially} rows")
+    df = cachedDF.generate_dataframe_timestamp_indexed(num_rows_initially, total_number_rows, cachedDF.TIME_UNIT)
+    dataframe_dump_to_log("df", df)
+    empty_df = df.drop(df.index)
+    dataframe_dump_to_log("empty", empty_df)
+    total_number_rows = total_number_rows + num_rows_initially
+    lib.write(symbol, data=df, prune_previous_versions=True)
+    cachedParts = CacheParts(df)
```

Comment:
I don't understand the need for `CacheParts`. Can't we just create a dataframe, stage chunks from it, finalize it, and check that the result matches the original dataframe?

Reply:
We could .... if all we want to stage is small dataframes that fit in memory. Once we produce more DFs that we store in symbol than what we can store potentially in memory cache that will not be possible.

Thus if we leave the decision of what parts of DF to cache only and later check to such class it will be easier to create test with much more data. 

While this might not have be the concrete case - in many other occasions it could be.

I have no problem to remove this now and use direct dataframes cached and appeneded in huge one this time, and use this approach when we really lack enough mem